tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -0,0 +1,121 @@
+import eu.timepit.refined.auto._
+import jakarta.inject.Inject
+import org.apache.james.jmap.api.model.{IdentityId, UploadId, UploadMetaData}
+import org.apache.james.jmap.api.upload.UploadRepository
+import org.apache.james.jmap.core.CapabilityIdentifier.{CapabilityIdentifier, JMAP_CORE}
+import org.apache.james.jmap.core.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.core.{AccountId, Invocation, SessionTranslator, UuidState}
+import org.apache.james.jmap.method.{InvocationWithContext, MethodRequiringAccountId}
+import org.apache.james.jmap.routes.SessionSupplier
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.metrics.api.MetricFactory
+import org.reactivestreams.Publisher
+import play.api.libs.json.{JsError, JsObject, JsSuccess}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class PublicAssetSetMethod @Inject()(val createPerformer: PublicAssetSetCreatePerformer,
+                                     val metricFactory: MetricFactory,
+                                     val sessionTranslator: SessionTranslator,
+                                     val sessionSupplier: SessionSupplier) extends MethodRequiringAccountId[PublicAssetSetRequest] {
+
+  override val methodName: Invocation.MethodName = MethodName("PublicAsset/set")
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, LINAGORA_PUBLIC_ASSETS)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: PublicAssetSetRequest): Publisher[InvocationWithContext] =
+    for {
+      created <- createPerformer.createPublicAssets(mailboxSession, request)
+      // Add updated, notUpdated, destroyed, notDestroyed here when they are implemented
+    } yield {
+      InvocationWithContext(
+        invocation = Invocation(
+          methodName = methodName,
+          arguments = Arguments(
+            PublicAssetSerializer.serializePublicAssetSetResponse(PublicAssetSetResponse(
+              accountId = request.accountId,
+              oldState = Some(UuidState.INSTANCE),
+              newState = UuidState.INSTANCE,
+              created = Some(created.retrieveCreated).filter(_.nonEmpty),
+              notCreated = Some(created.retrieveErrors).filter(_.nonEmpty))).as[JsObject]),
+          methodCallId = invocation.invocation.methodCallId),
+        processingContext = invocation.processingContext)
```

Comment:
Did we record the PublicAsset into the invocation processing context?

Ie and I refer it in JMAP via `#creationId` in the same JMAP call?